app/code/community/Lesti/Fpc/Helper/Data.php

```diff
@@ -258,4 +259,30 @@ public function getContentType(\Mage_Core_Controller_Response_Http $response)
 
         return 'text/html; charset=UTF-8';
     }
+
+    /**
+     * Transform array parameter to string to avoid error and make correct validation in regexp pattern
+     *
+     * @param mixed $param Input parameter
+     *
+     * @return mixed
+     */
+    protected function _paramToString($param)
+    {
+        if (is_array($param)) {
+            $param = array_reduce(
+                $param,
+                function ($carry, $item) {
```

Comment:
Hello, 
Which conflicts? It just a line of text.
If you will use serialize() or json_encode(), you will get incorrect value in preg_match() section.

for example, when you define at BO expression for some field ^[0-9]+$ (you expect some id(s)),  then:
1) with json_encode you will receive "[25,98]" string, and it will be incorrect, because does not match our pattern
2) with reducing array to string (via concatenation) you will receive "2598" string and it will be always correct, even if you will send multidimensional array in parameter

In your code this is simple can_cache flag and data checks only for valid format, but its context not necessary.